mtags/src/main/scala/scala/meta/internal/pc/Completions.scala

```diff
@@ -669,6 +676,42 @@ trait Completions { this: MetalsGlobal =>
     }
   }
 
+  /**
+   * Find a method definition right after the given position.
+   *
+   * @param pos The position of scaladoc in the original source.
+   *            This class will find the associated member def based on this pos.
+   */
+  class AssociatedMemberDefFinder(pos: Position) extends Traverser {
+    private var defs: List[MemberDef] = Nil
+
+    /**
+     * Collect all the member definitions whose position is
+     * below the given `pos`. And then return the closest member definiton.
+     */
+    def findAssociatedDef(root: Tree): Option[MemberDef] = {
+      defs = Nil
+      traverse(root)
+      defs.sortBy(_.pos.point).headOption
+    }
+    override def traverse(t: Tree): Unit = {
+      t match {
+        case typedef @ TypeDef(_, _, _, _) => process(typedef)
+        case clsdef @ ClassDef(_, _, _, _) => process(clsdef)
+        case defdef @ DefDef(_, _, _, _, _, _) => process(defdef)
+        case moduledef @ ModuleDef(_, _, _) => process(moduledef)
+        case pkgdef @ PackageDef(_, _) => process(pkgdef)
+        case valdef @ ValDef(_, _, _, _) => process(valdef)
+        case _ if treePos(t).includes(pos) => super.traverse(t)
```

Comment:
What we construct scaladoc for is just `ClassDef` and `DefDef`, but collecting all the `MemberDef`  to avoid the following pattern.

```scala
/**@@
type Foo = Int // If we don't collect TypeDef

def foo(x: Int): Int = ???
```

to

```scala
/**
   *
   * @param x
   * @return
   */
type Foo = Int // If we don't collect TypeDef

def foo(x: Int): Int = ???
```